modules/lineDetector/src/main.cpp

```diff
@@ -886,43 +900,38 @@ class Detector : public RFModule, public lineDetector_IDL
     }
 
     /****************************************************************/
-    bool getCameraOptions(cv::Mat &cam_intrinsic, cv::Mat &cam_distortion)
+    bool getCameraOptions(cv::Mat &cam_intrinsics, cv::Mat &cam_distortion)
     {
-        if (camPort.getOutputCount()>0)
-        {
-            yInfo()<<"Reading intrinsics from camera";
-            Bottle cmd,rep;
-            cmd.addVocab(VOCAB_RGB_VISUAL_PARAMS);
-            cmd.addVocab(VOCAB_GET);
-            cmd.addVocab(VOCAB_INTRINSIC_PARAM);
-            if (camPort.write(cmd,rep))
-            {
-                if (rep.size()>=4)
-                {
-                    yarp::os::Bottle *intrinsics = rep.get(3).asList();
-                    fx = intrinsics->find("focalLengthX").asDouble();
-                    fy = intrinsics->find("focalLengthY").asDouble();
-                    px = intrinsics->find("principalPointX").asDouble();
-                    py = intrinsics->find("principalPointY").asDouble();
-
-                    yInfo()<<"Camera intrinsics:"<<fx<<fy<<px<<py;
-
-                    cam_intrinsic.at<double>(0,0)=fx;
-                    cam_intrinsic.at<double>(0,2)=px;
-                    cam_intrinsic.at<double>(1,1)=fy;
-                    cam_intrinsic.at<double>(1,2)=py;
-                    cam_intrinsic.at<double>(2,2)=1.0;
-
-                    cam_distortion.at<double>(0,0)=0.0; //k1;
-                    cam_distortion.at<double>(0,1)=0.0; //k2;
-                    cam_distortion.at<double>(0,2)=0.0; //t1;
-                    cam_distortion.at<double>(0,3)=0.0; //t2;
-                    cam_distortion.at<double>(0,4)=0.0; //k3;
-
-                    return true;
-                }
+        yarp::dev::IRGBDSensor* irgbd;
+        rgbdDrv.view(irgbd);
```

Comment:
If I understand correctly, `getCameraOptions` is called only once... If this is not the case, perhaps you should move the view in the constructor (and of course save the pointer somewhere), in order to avoid calling this every time the method is called.

Reply:
Yes, it's called only once and thus I decided to go for a local variable and declare it close to where it's used.